### PR TITLE
Updated analysis: missing NBL samples in histologies file

### PR DESCRIPTION
**Pull request by zzgeng**
#### What scientific question is your analysis addressing?
NBL subtyping results are missing in the histologies file. 

#### What was your approach?
Update 01 R script in molecular-subtyping-pathology module and re-run the molecular-subtyping-pathology and molecular-subtyping-integrate modules. 

#### What GitHub issue does your pull request address?
https://github.com/PediatricOpenTargets/ticket-tracker/issues/550

#### Which areas should receive a particularly close look?
analyses/molecular-subtyping-pathology/01-compile-subtyping-results.Rmd
analyses/molecular-subtyping-pathology/results
analyses/molecular-subtyping-integrate/results/histologies.tsv

#### Is there anything that you want to discuss further?
No

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?
Yes

#### What types of results are included (e.g., table, figure)?
Table

#### What is your summary of the results?
Add NBL subtyping results to histologies file

#### Reproducibility Checklist

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [x] This analysis has been added to continuous integration.

#### Documentation Checklist

- [x] This analysis module has a `README` and it is up to date.
- [x] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [x] The analytical code is documented and contains comments.

### CODE REVIEW

**Review by atlas4213 — APPROVED** [review]
The module runs and I see the changes in the input files but just from an understanding point I see in 01-integrate-subtyping.nb.html module that there are what looks to be hashes being added into the code. What is that doing?

**Review by atlas4213 — APPROVED** [review]
The module runs and I see the changes in the input files but just from an understanding point I see in 01-integrate-subtyping.nb.html module that there are what looks to be hashes being added into the code. What is that doing?

**Review by atlas4213 — APPROVED** [review]
The module runs and I see the changes in the input files but just from an understanding point I see in 01-integrate-subtyping.nb.html module that there are what looks to be hashes being added into the code. What is that doing?

**Comment by jharenza** [comment]
> The module runs and I see the changes in the input files but just from an understanding point I see in 01-integrate-subtyping.nb.html module that there are what looks to be hashes being added into the code. What is that doing?

This is just the html file which often will frequently change with a rerun. In the case of 01 script, it looks like two columns of the histology file's classes changed, and although the script did not change, that change reflects in the html rendering and some of the hashing changes. It is expected behavior, although I am not sure the column class changes are! @zzgeng can you confirm these changes (I see them in a few scripts).

File: analyses/molecular-subtyping-pathology/results/glialneuronal_tumor_subtypes.tsv [hunk 1]
```diff
@@ -2,38 +2,38 @@ Kids_First_Participant_ID	sample_id	Kids_First_Biospecimen_ID	broad_histology	sh
...
-PT_DYVAW1K2	7316-1112	BS_75P5BC58	Neuronal and mixed neuronal-glial tumor	GNT	Glial-neuronal tumor,  RTK	GNT, RTK	LGG, FGFR	Updated via OpenPedCan subtyping and pathology_free_text_diagnosis	Low-grade glioma/astrocytoma (WHO grade I/II)	Initial CNS Tumor	glioneuronal tumor	Glial-neuronal tumor NOS
-PT_DYVAW1K2	7316-1112	BS_WXT1VK25	Neuronal and mixed neuronal-glial tumor	GNT	Glial-neuronal tumor,  RTK	GNT, RTK	LGG, FGFR	Updated via OpenPedCan subtyping and pathology_free_text_diagnosis	Low-grade glioma/astrocytoma (WHO grade I/II)	Initial CNS Tumor	glioneuronal tumor	Glial-neuronal tumor NOS
+PT_DYVAW1K2	7316-1112	BS_75P5BC58	Neuronal and mixed neuronal-glial tumor	GNT	Glial-neuronal tumor,  RTK	GNT, RTK	NA	Updated via OpenPedCan subtyping and pathology_free_text_diagnosis	Low-grade glioma/astrocytoma (WHO grade I/II)	Initial CNS Tumor	glioneuronal tumor	Glial-neuronal tumor NOS
```
Comment (jharenza) [inline analyses/molecular-subtyping-pathology/results/glialneuronal_tumor_subtypes.tsv]: investigate whether this is expected

**Review by sangeetashukla — COMMENTED** [review]
Hi @zzgeng Please tag me in a comment when you have reviewed the files in this branch as we discussed. 
Then I am ready to approve.

**Comment by zzgeng** [comment]
> 

Hi @sangeetashukla , I just push a commit to re-run the module with the latest updates. I also tried to repeat this PR in a new branch and the results are reproducible. Please let me know you have any question.

**Review by sangeetashukla — COMMENTED** [review]
Hi @zzgeng I reviewed the changes, and the result files in the PR and re-ran the `molecular-subtyping-integrate` and `molecular-subtyping-pathology` modules locally.
However, I see a `molecular-subtyping-pathology/results/glialneuronal_tumor_subtypes.tsv` generated, which is not in this PR. 

Can you please review/confirm ?

**Comment by zzgeng** [comment]
> Hi @zzgeng I reviewed the changes, and the result files in the PR and re-ran the `molecular-subtyping-integrate` and `molecular-subtyping-pathology` modules locally. However, I see a `molecular-subtyping-pathology/results/glialneuronal_tumor_subtypes.tsv` generated, which is not in this PR.
> 
> Can you please review/confirm ?

Hello @sangeetashukla , sorry for the late reply. I checked the changes in `molecular-subtyping-pathology/results/glialneuronal_tumor_subtypes.tsv` in previous commit, which mainly happened in the `molecular_subtype_methyl` in LGG sample. Due to the updates, these changes were gone after I re-run the file.

**Comment by jharenza** [comment]
hi @zzgeng I have updated the GNT subtyping since this PR, so this needs to be rerun once more. I think it requires EC2, but I had not had time to set it up yet..

**Comment by zzgeng** [comment]
> hi @zzgeng I have updated the GNT subtyping since this PR, so this needs to be rerun once more. I think it requires EC2, but I had not had time to set it up yet..

Do I need to run any other modules than 'molecular_subtyping_pathology' and 'molecular_subtyping_integrate' for this PR?

**Comment by jharenza** [comment]
No, the GNT is in the pathology module

**Review by jharenza — APPROVED** [review]
this looks good to me now, thank you!

**Pull request merged**